File: examples/python/raspberry/logs.py
import grafana_foundation_sdk.models.dashboard as dashboard
from grafana_foundation_sdk.cog import builder as cogbuilder
from .common import default_logs, basic_loki_query


def errors_in_system_logs() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_logs()
        .title("Errors in system logs")
        .with_target(basic_loki_query('{level=~"err|crit|alert|emerg", job="integrations/raspberrypi-node", instance="$instance"}'))
        .with_target(basic_loki_query('{filename=~"/var/log/syslog*|/var/log/messages*", job="integrations/raspberrypi-node", instance="$instance"} |~".+(?i)error(?-i).+"'))
    )


def auth_logs() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_logs()
        .title("Auth logs")
        .with_target(basic_loki_query('{unit="ssh.service", job="integrations/raspberrypi-node", instance="$instance"}'))
        .with_target(basic_loki_query('{filename=~"/var/log/auth.log|/var/log/secure", job="integrations/raspberrypi-node", instance="$instance"}'))
    )


def kernel_logs() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_logs()
        .title("Kernel logs")
        .with_target(basic_loki_query('{transport="kernel", job="integrations/raspberrypi-node", instance="$instance"}'))
        .with_target(basic_loki_query('{filename="/var/log/kern.log", job="integrations/raspberrypi-node", instance="$instance"}'))
    )


def all_system_logs() -> cogbuilder.Builder[dashboard.Panel]:
    return (
        default_logs()
        .title("All system logs")
        .with_target(basic_loki_query('{transport!="", job="integrations/raspberrypi-node", instance="$instance"}'))
        .with_target(basic_loki_query('{filename!="", job="integrations/raspberrypi-node", instance="$instance"}'))
    )

File: examples/typescript_dashboardv2/logs.ts
import {basicLokiQuery, defaultLogs} from "./common";
import {PanelBuilder, QueryGroupBuilder, TargetBuilder} from "../../generated/typescript/src/dashboardv2beta1";

export const errorsInSystemLogs = (): PanelBuilder => {
    return new PanelBuilder()
        .title("Errors in system logs")
        .visualization(defaultLogs())
        .data(new QueryGroupBuilder().targets([
            new TargetBuilder().query(basicLokiQuery(`{level=~"err|crit|alert|emerg", job="integrations/raspberrypi-node", instance="$instance"}`).refId("A")),
            new TargetBuilder().query(basicLokiQuery(`{filename=~"/var/log/syslog*|/var/log/messages*", job="integrations/raspberrypi-node", instance="$instance"} |~".+(?i)error(?-i).+"`).refId("B")),
        ]));
};

export const authLogs = (): PanelBuilder => {
    return new PanelBuilder()
        .title("Auth logs")
        .visualization(defaultLogs())
        .data(new QueryGroupBuilder().targets([
            new TargetBuilder().query(basicLokiQuery(`{unit="ssh.service", job="integrations/raspberrypi-node", instance="$instance"}`).refId("A")),
            new TargetBuilder().query(basicLokiQuery(`{filename=~"/var/log/auth.log|/var/log/secure", job="integrations/raspberrypi-node", instance="$instance"}`).refId("B")),
        ]));
};

export const kernelLogs = (): PanelBuilder => {
    return new PanelBuilder()
        .title("Kernel logs")
        .visualization(defaultLogs())
        .data(new QueryGroupBuilder().targets([
            new TargetBuilder().query(basicLokiQuery(`{transport="kernel", job="integrations/raspberrypi-node", instance="$instance"}`).refId("A")),
            new TargetBuilder().query(basicLokiQuery(`{filename="/var/log/kern.log", job="integrations/raspberrypi-node", instance="$instance"}`).refId("B")),
        ]));
};

export const allSystemLogs = (): PanelBuilder => {
    return new PanelBuilder()
        .title("All system logs")
        .visualization(defaultLogs())
        .data(new QueryGroupBuilder().targets([
            new TargetBuilder().query(basicLokiQuery(`{transport!="", job="integrations/raspberrypi-node", instance="$instance"}`).refId("A")),
            new TargetBuilder().query(basicLokiQuery(`{filename!="", job="integrations/raspberrypi-node", instance="$instance"}`).refId("B")),
        ]));
};
